jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/XdgDirectories.java

```diff
@@ -0,0 +1,174 @@
+      Paths.get("google-cloud-tools-java").resolve("jib");
+  private static final Path JIB_SUBDIRECTORY_OTHER = Paths.get("Google").resolve("Jib");
+
+  public static Path getCacheHome() {
+    return getCacheHome(System.getProperties(), System.getenv());
+  }
+
+  public static Path getConfigHome() {
+    return getConfigHome(System.getProperties(), System.getenv());
+  }
+
+  /**
+   * Returns {@code $XDG_CACHE_HOME}, if available, or resolves the OS-specific user cache home
+   * based.
+   *
+   * <p>For Linux, this is {@code $HOME/.cache/}.
+   *
+   * <p>For Windows, this is {@code %LOCALAPPDATA%}.
+   *
+   * <p>For macOS, this is {@code $HOME/Library/Application Support/}.
+   */
+  @VisibleForTesting
+  static Path getCacheHome(Properties properties, Map<String, String> environment) {
+    // Use environment variable $XDG_CACHE_HOME if set and not empty.
+    String xdgCacheHome = environment.get("XDG_CACHE_HOME");
+    if (xdgCacheHome != null && !xdgCacheHome.trim().isEmpty()) {
+      return Paths.get(xdgCacheHome);
+    }
+
+    String userHome = properties.getProperty("user.home");
+    Path xdgPath = Paths.get(userHome, ".cache");
+
+    String rawOsName = properties.getProperty("os.name");
+    String osName = rawOsName.toLowerCase(Locale.ENGLISH);
+
+    if (osName.contains("linux")) {
+      return xdgPath;
+
+    } else if (osName.contains("windows")) {
+      // Use %LOCALAPPDATA% for Windows.
+      String localAppDataEnv = environment.get("LOCALAPPDATA");
+      if (localAppDataEnv == null || localAppDataEnv.trim().isEmpty()) {
+        LOGGER.warning("LOCALAPPDATA environment is invalid or missing");
+        return xdgPath;
+      }
+      Path localAppData = Paths.get(localAppDataEnv);
+      if (!Files.exists(localAppData)) {
+        LOGGER.warning(localAppData + " does not exist");
+        return xdgPath;
+      }
+      return localAppData;
+
+    } else if (osName.contains("mac") || osName.contains("darwin")) {
+      // Use '~/Library/Application Support/' for macOS.
+      Path applicationSupport = Paths.get(userHome, "Library", "Application Support");
+      if (!Files.exists(applicationSupport)) {
+        LOGGER.warning(applicationSupport + " does not exist");
+        return xdgPath;
+      }
+      return applicationSupport;
+    }
+
+    throw new IllegalStateException("Unknown OS: " + rawOsName);
+  }
+
+  /**
+   * Returns config directory based on {@code $XDG_CONFIG_HOME}, if available, or resolves the
+   * OS-specific user config directory.
+   *
+   * <p>For Linux, this is {@code $HOME/.config/google-cloud-tools-java/jib/}.
+   *
+   * <p>For Windows, this is {@code %LOCALAPPDATA%\Google\Jib/}.
+   *
+   * <p>For macOS, this is {@code $HOME/Library/Preferences/Google/Jib/}.
+   */
+  @VisibleForTesting
+  static Path getConfigHome(Properties properties, Map<String, String> environment) {
+    Path windowsSubDirectory = JIB_SUBDIRECTORY_OTHER.resolve("Config");
```

Comment:
Why not having `windowsSubDirectory` as a constant?

Reply:
It's different for cache and config.